### ISSUE TITLE
go.testEnvVars should override go.testEnvFile

### ISSUE THREAD

**Issue by gtalarico** [microsoft/vscode-go#2398]
I haven't found docs around usage of `go.testEnvFile` and  `go.testEnvVars`, 
but I instinctively expected them to behave similarly to how `env` and `envFile` work in `launch.js`: Variables are loaded from `.env`, but if explicitly set by `testEnvVars`, this value will prevail over `.env` value

https://code.visualstudio.com/docs/nodejs/nodejs-debugging#_load-environment-variables-from-external-file-node
> Any environment variable specified in the env dictionary will override variables loaded from the file.

# To Reproduce

```bash
# .env
FOO=1
BAR=2
```

```json
# .vscode/settings.json
{
    "go.testEnvFile": "${workspaceFolder}/.env",
    "go.testEnvVars": {
        "FOO": "BACON",
        "BAR": "EGGS",
    },
    "go.testFlags": ["-v"]
}
```

#### main_test.go
```go
package main

import (
	"fmt"
	"os"
	"testing"
)

func TestEnvVar(t *testing.T) {
	foo := os.Getenv("FOO")
	bar := os.Getenv("BAR")
	fmt.Printf("foo: %v \n", foo)
	fmt.Printf("bar: %v \n", bar)

}
```

## Result

```
=== RUN   TestEnvVar
foo: 1 
bar: 2 
--- PASS: TestEnvVar (0.00s)
PASS
ok  	github.com/weconnect/playground	(cached)
Success: Tests passed.
```

* Click `Run Test` in vscode
* Expected: FOO and BAR = "BACON" and "EGGS"
* Got: FOO and BAR = "1" and "2"

**Comment by coopermaruyama** [comment]
100% agree with this.

In our setup, we check in environment variables for testing in a `.env.test` file which is used with `go.testEnvVars` which has some sane defaults, and then (ideally) developers would be able to override them with `go.testEnvVars` if needed. 

However, because of the load order (as described above), this is not possible for us. I really think `go.testEnvVars` should take precedence given that it seems to be the ideal location for user-space fine tuning.

**Comment by ramya-rao-a** [comment]
This change is now available in the latest update (0.11.1) to the Go extension

Thanks for the PR @coopermaruyama and thanks for reporting @gtalarico!
